fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt

```diff
@@ -0,0 +1,106 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.FloatingActionButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Floating action button, to be used inside a [Box], to allow aligning the FAB to the bottom end corner.
+ *
+ * @param modifier [Modifier] to be applied to the action button.
+ * @param icon [Painter] icon to be displayed inside the action button.
+ * @param label Text to be displayed next to the icon.
+ * @param onClick Invoked when the button is clicked.
+ */
+@Composable
+fun FloatingActionButton(
+    modifier: Modifier = Modifier,
+    icon: Painter,
+    label: String? = null,
+    onClick: () -> Unit,
+) {
+    FloatingActionButton(
+        onClick = onClick,
+        modifier = Modifier.then(modifier),
+        backgroundColor = FirefoxTheme.colors.actionPrimary,
+        contentColor = FirefoxTheme.colors.textActionPrimary,
+        elevation = FloatingActionButtonDefaults.elevation(99.dp),
+    ) {
+        Row(
+            modifier = Modifier
+                .wrapContentSize()
+                .padding(16.dp)
+                .animateContentSize(),
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Icon(
+                painter = icon,
+                contentDescription = if (Theme.getTheme() == Theme.Private) {
+                    stringResource(
+                        R.string.add_private_tab,
+                    )
+                } else {
+                    stringResource(
+                        R.string.add_tab,
+                    )
+                },
+                tint = FirefoxTheme.colors.iconOnColor,
+            )
+
+            if (!label.isNullOrBlank()) {
+                Row {
+                    Spacer(Modifier.width(12.dp))
+
+                    Text(
+                        text = label,
+                        style = FirefoxTheme.typography.button,
+                        maxLines = 1,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@LightDarkPreview
+@Composable
+private fun FloatingActionButtonPreview() {
+    val label = remember { mutableStateOf<String?>("LABEL") }
```

Comment:
```suggestion
    var label by remember { mutableStateOf("LABEL") }
```

just a small syntax change to make it a little more clean.